Front-End/src/Components/SignUp.js

```diff
@@ -1,14 +1,72 @@
-import React from "react";
 import { Link, useNavigate } from "react-router-dom";
+import React from "react";
+
 import "../CSS/SignUp.css";
 import Gif from "../Resources/Chalk_Shapian.gif";
 
 const SignUp = () => {
   const navigate = useNavigate();
-  const signup = () => {
-    localStorage.setItem('LoggedIn',true);
-    navigate("/Discover");
-  }
+  const signup = async (event) => {
+    event.preventDefault();
+  
+    const chalkName = document.getElementById("chalkNameSignUp").value;
+    const name = document.getElementById("nameSignUp").value;
+    const email = document.getElementById("emailSignUp").value;
+    const password = document.getElementById("passwordSignUp").value;
+    const confirmPassword = document.getElementById(
+      "confirmPasswordSignUp"
+    ).value;
+  
+    // Check if any of the input fields are empty
+    if (!chalkName || !name || !email || !password || !confirmPassword) {
+      alert("Please fill out all the fields before submitting.");
+      return;
+    }
+  
+    if (password !== confirmPassword) {
+      alert("Passwords do not match");
+      return;
+    }
+  
+    // Password must be at least 8 characters long and contain at least one uppercase letter and one special character
+    const passwordRegex = /^(?=.*[A-Z])(?=.*[!@#$&*])(?=.{8,})/;
+    if (!passwordRegex.test(password)) {
+      alert(
+        "Password must be at least 8 characters long and contain at least one uppercase letter and one special character"
+      );
+      return;
+    }
+  
+    try {
+      const response = await fetch("http://localhost:6969/api/user/", {
```

Comment:
Add `http://localhost:6969` to `.env` and use here using`process.env`.

Reply:
done